addons/stock_account/models/product.py

```diff
@@ -849,22 +849,23 @@ class ProductCategory(models.Model):
 
     @api.model
     def _get_stock_account_property_field_names(self):
```

Comment:
I guess we can remove the override of `write` in 
https://github.com/odoo/odoo/blob/cb97c0b47cfdb7e668ff058263813e5d56ede330/addons/mrp_account/models/product.py#L125
And  `_get_stock_account_property_field_names` can always append `'property_stock_account_production_cost_id'`
https://github.com/odoo/odoo/blob/cb97c0b47cfdb7e668ff058263813e5d56ede330/addons/mrp_account/models/product.py#L128
because I think the those codes were added just to avoid `property_stock_account_production_cost_id` to be modified in the constains, since the `write in constrains` doesn't exist, those codes are useless

Reply:
ok to remove the write override but property_stock_account_production_cost_id is the only one that can be `False` even in real time. That's why I introduced  `_get_mandatory_stock_account_property_field_names()`

Reply:
Yes, I see the reason you introduce `_get_mandatory_stock_account_property_field_names()`
but thanks to that, the override of `def _get_stock_account_property_field_names(self): ` 
https://github.com/odoo/odoo/blob/cb97c0b47cfdb7e668ff058263813e5d56ede330/addons/mrp_account/models/product.py#L128
should be simplified by always appending `'property_stock_account_production_cost_id'` which won't be checked in the new constrain